OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/OptimoveConfig.java

```diff
@@ -132,6 +168,72 @@ private void setMinLogLevel(@Nullable LogLevel minLogLevel){
         this.minLogLevel = minLogLevel;
     }
 
+    void setCredentials(@Nullable String optimoveCredentials, @Nullable String optimobileCredentials){
+        if (optimoveCredentials == null && optimobileCredentials == null) {
+            throw new IllegalArgumentException("Should provide at least optimove or optimobile credentials");
+        }
+
+        if (optimoveCredentials != null && !this.featureSet.has(FeatureSet.Feature.OPTIMOVE)){
+            throw new IllegalArgumentException("Cannot set credentials for optimove as it is not in the desired feature set");
+        }
+
+        if (optimobileCredentials != null && !this.featureSet.has(FeatureSet.Feature.OPTIMOBILE)){
+            throw new IllegalArgumentException("Cannot set credentials for optimobile as it is not in the desired feature set");
+        }
+
+        if (this.hasFinishedInitialisation()){
+            throw new IllegalStateException("OptimoveConfig: credentials are already set");
```

Comment:
Is it preferable that this is a warn log perhaps?

Is IllegalStateException checked or will it just crash the app?

Perhaps acceptable, maybe we should document on the method.

Reply:
exception is unchecked. this is debatable

a) could be as strict as possible to enforce correct usage, e.g. what if try to set multiple times and different. Perhaps unexpected that 2nd set would do nothing
b) unlikely they will set different, safer is to not crash the app

maybe b is better indeed

Reply:
Just querying, I do personally like trying to enforce 'correct' usage, but we have had feedback that sometimes crashing is unexpected as testing doesn't adequately identify some edge cases.

@eligutovsky @k-antipochkin thoughts on this? I'm keen that we align between the SDKs too for a consistent DX & easier reasoning about behaviour amongst the team.

Reply:
Seems like a case where crashes are a friend. Maybe worth crashing if credentials changed?

Reply:
👌